src/aiori-HDF5.c

```diff
@@ -99,37 +103,60 @@ static int HDF5_RmDir(const char *, IOR_param_t *);
 static int HDF5_Access(const char *, int, IOR_param_t *);
 static int HDF5_Stat(const char *, struct stat *, IOR_param_t *);
 
+static int set_options(IOR_param_t *);
+static void InjectFault(IOR_param_t *);
+
 /************************** O P T I O N S *****************************/
 typedef struct{
   int collective_md;
   int chunk_size;
+  char *daos_obj_class;
+  unsigned int numb_faults;
+  char *server_ranks_str;
+  char *fault_iterates_str;
+  char *fault_rw_str;
 } HDF5_options_t;
+
+int *daos_server_ranks = NULL;
+int *fault_iterates = NULL;
+int *fault_rw = NULL;
 /***************************** F U N C T I O N S ******************************/
 
 static option_help * HDF5_options(void ** init_backend_options, void * init_values){
+  int i;
   HDF5_options_t * o = malloc(sizeof(HDF5_options_t));
 
   if (init_values != NULL){
     memcpy(o, init_values, sizeof(HDF5_options_t));
-  }else{
+  } else{
     /* initialize the options properly */
     o->collective_md = 0;
     o->chunk_size = 0;
+    o->daos_obj_class = strdup("S1");
```

Comment:
init to NULL rather than strdup; then before you call the set function, check if NULL and pass a SX if it is.

Reply:
I changed it.